hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMRatisServerImpl.java

```diff
@@ -338,6 +340,49 @@ public boolean addSCM(AddSCMRequest request) throws IOException {
     }
   }
 
+  public void updateRaftPeerPriority(String peerId) throws IOException {
+    List<RaftPeer> raftPeers =
+        new ArrayList<>(getDivision().getGroup().getPeers());
+    List<RaftPeer> oldPeers = new ArrayList<>(raftPeers);
+    RaftPeer peerToUpdate = raftPeers.stream()
+        .filter(peer -> peer.getId().toString().equals(peerId)).findFirst()
+        .get();
+    RaftPeer newPeer = RaftPeer.newBuilder(peerToUpdate).setPriority(1).build();
+    raftPeers.add(newPeer);
+    raftPeers.remove(peerToUpdate);
+    final SetConfigurationRequest configRequest =
+        new SetConfigurationRequest(clientId, division.getInfo().getLeaderId(),
+            division.getGroup().getGroupId(), nextCallId(), raftPeers);
+    RaftClient raftClient = null;
+    try {
+      RaftClient.Builder clientBuilder = RaftClient.newBuilder().setRaftGroup(
+              RaftGroup.valueOf(division.getGroup().getGroupId(), oldPeers))
+          .setProperties(new RaftProperties());
+      boolean success = false;
+      int retries = 0;
+      raftClient = clientBuilder.build();
+      while (!success && retries < UPDATE_PEER_PRIORITY_RETRY_COUNT) {
```

Comment:
We don't need a retry logic for updating the priority. If this call fails, it means that this SCM is not able to communicate with other SCMs. So there is no point in updating the priority of this SCM.